include/seqan3/argument_parser/detail/format_parse.hpp

```diff
@@ -301,8 +299,37 @@ class format_parse : public format_base
         stream >> value;
 
         if (stream.fail() || !stream.eof())
+        {
             throw type_conversion_failed("Argument " + in + " could not be casted to type " +
                                          get_type_name_as_string(value) + ".");
+        }
+    }
+
+    /*!\brief Tries to cast an input string into a value.
+     * \tparam option_t Must model seqan3::string_convertible.
+     * \param[out] value Stores the cast value.
+     * \param[in]  in    The input argument to be cast.
+     *
+     * \throws seqan3::type_conversion_failed
+     */
+    template <typename option_t>
+    //!\cond
+        requires string_convertible<option_t>
+    //!\endcond
+    void retrieve_value(option_t & value, std::string const & in)
```

Comment:
why an out-parameter and not a return value?

Reply:
Because this is an overload that is from quite old code :see_no_evil: 
I don't remember if there is a specific reason but to refactor the 3 other function would exceed the scope of this PR.

Reply:
Yeah, doesn't have to happen here, but I stumbled over it and we have a clear policy for returning by value (unless there is a reason not to).

Reply:
I'll add a card

Reply:
although the function then is still a template because I need to know the type 
soo the call would look like 
```
op = retrieve_value<option_t>(input_str)
```